mobile/src/libs/packages/store/middlewares/middlewares.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  type AnyAction,
+  type Dispatch,
+  type Middleware,
+} from '@reduxjs/toolkit';
+import { isRejected } from '@reduxjs/toolkit';
+
+import { NotificationService } from '#libs/packages/services/notification-service';
+
+const notificationService = new NotificationService();
+
+const handleAction =
+  (next: Dispatch): ((action: AnyAction) => AnyAction) =>
+  (action: AnyAction): AnyAction => {
+    if (isRejected(action) && action.error.message) {
+      notificationService.error(action.error.message);
```

Comment:
Please check comments related to error handling in web app please

Reply:
Hi @v0ldemar01.
Could you help me with the changes requested? from https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/pull/35 that are related to this comment.
1. `Let's proceed with only handleError without handleAction` if I just use handleError like this 
``` javascript
 const handleError: Middleware =
  () => (next: Dispatch) => (action: AnyAction) => {
    const result = next(action);

    if (isRejected(action)) {
      const errorMessage = action.error.message as string;
      notificationService.error(errorMessage);
    }

    return result;
  };
```
I got a lint error 
`  13:26  error  Move arrow function to the outer scope  unicorn/consistent-function-scoping`

2. `Use dispatch to call action that contains NotificationService.error(action.error.message);` I don't really understand how I can handle this.

3. `Use curly and return for declaration` I don't understand what you mean here 

Thanks in advance. 😃

Reply:
Hey @BrianSammit. I will begin from down to up. Using curly and return means that instead of applying `const func = () => () => () => {}` prefer
```js
const func = () => {
    return () => {
        return {}
    }
}
```
It would be useful for increasing the readness of a code. Currently we have a situation that prettier moves each block into new line.
2. Let's get a dispatch function and call action for example
```js
const { message } = action.error;
dispatch(appActions.notify({ type: NotificationType.ERROR, message }));
```
Action `notify` will execute `notificationService[type](message);`
3. This one is related to the first one. So it would be solved